frontend_v2/src/app/components/challenge/challengeanalytics/challengeanalytics.component.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+// import service
+import { ApiService } from '../../../services/api.service';
+import { GlobalService } from '../../../services/global.service';
+import { EndpointsService } from '../../../services/endpoints.service';
+import { WindowService } from '../../../services/window.service';
+import { ChallengeService } from '../../../services/challenge.service';
+
+@Component({
+  selector: 'app-challengeanalytics',
+  templateUrl: './challengeanalytics.component.html',
+  styleUrls: ['./challengeanalytics.component.scss'],
+})
+export class ChallengeanalyticsComponent implements OnInit {
+  /**
+   * Challenge object
+   */
+  currentChallengeDetails: any;
+  challengeId = null;
+  currentPhase = [];
+  totalSubmission = {};
+  totalParticipatedTeams = {};
+  lastSubmissionTime = {};
+  totalChallengeTeams = [];
+  routePath = '/auth/login';
```

Comment:
Can you please use string docs for all the variables like we have used in other files @Kajol-Kumari